angr/analyses/cfg/cfg_fast.py

```diff
@@ -625,6 +628,11 @@ def __init__(self,
             regions = new_regions
         if not regions and self.project.arch.name != 'Soot':
             raise AngrCFGError("Regions are empty or all regions are skipped. You may want to manually specify regions.")
+        if extra_executable_regions:
+            for start_, end_ in extra_executable_regions:
+                if self._should_add_executable_region(start_) and (start_, end_) not in regions:
+                    regions.append((start_, end_))
```

Comment:
Question: Why don't you reuse `regions` by putting tuples in `extra_executable_regions` into `regions` and passing it to `CFGFast.__init__()`? I don't see why we need to add a new argument here.

Reply:
```

Reply:
```
        regions = regions if regions is not None else self._executable_memory_regions(objects=objects,
                                                                                      force_segment=force_segment
                                                                                      )
```
If `regions` is set, `self._executable_memory_regions()` won't be execute. Thus, `regions` requires the whole list of executable region tuples. 

However, patcherex can only provide the start and end addresses of the added regions, not all the executable regions (according to my teammate, I'm not quite sure about the details)

In this case, in order to reuse `regions` I have to manually check the sections and segments. That's why I added extra_executable_regions to help automate the process.

Reply:
Correction: Patcherex is able to identify all the executable regions in a binary (including sections and segments). However, without knowing the value of `force_segment`, which is a parameter used when generating the CFG, patcherex is not able to generate the correct region tuple list (E.g. from executable sections OR from executable segments)

Reply:
I believe you should populate `regions` beforehand instead of adding another `extra_executable_regions` argument that is essentially duplicating what `regions` is meant to be.

I also disagree that you do not know the value of `force_segment` when running CFGFast(), because you (or whoever calls `project.analyses.CFGFast()` should specify the `force_segment` value!